api/src/main/java/org/openmrs/module/clientregistry/api/ClientRegistryManager.java

```diff
@@ -82,4 +92,26 @@ public void disableClientRegistry() {
 		
 		isRunning.set(false);
 	}
+
+	/**
+	 * Determine the appropriate PatientService class based off of the client registry transaction type configuration
+	 * @return PatientService class corresponding to the appropriate transaction type supported by the client registry
+	 * @throws IllegalArgumentException if defined transaction type is unsupported
+	 */
+	public ClientRegistryPatientService getPatientService() throws IllegalArgumentException {
```

Comment:
not sure i love this. maybe we move this logic to ClientRegistryActivator and only check the transaction type once on module startup (and on global property changed). ClientRegistryManager can hold a ClientRegistryPatientService variable with getters/setters and the provider would call ClientRegistryManager.getPatientService() without having to do the global property comparison each time? thoughts @pmanko ?